node/cloudtrail_enabled_all_regions-periodic.js

```diff
@@ -3,61 +3,76 @@
 //
 // Ensure CloudTrail is Enabled for All Regions
 // Description: Checks that a CloudTrail exists that is set to multi-region
-// 
+//
 // Trigger Type: Periodic
 // Required Parameter: None
 
-var aws  = require('aws-sdk');
+var aws = require('aws-sdk');
 var s3 = new aws.S3();
 var zlib = require('zlib');
 var config = new aws.ConfigService();
- 
-// This function checks whether CloudTrail is enabled with multi region trail in the home region. 
+
+// This function checks whether CloudTrail is enabled with multi region trail in the home region.
 // Do not use this rule in other regions.
- 
+
 function evaluateCompliance(configurationItems, ruleParameters, context) {
     checkDefined(configurationItems, "configurationItems");
     checkDefined(ruleParameters, "ruleParameters");
-    
+
     for (var i = 0; i < configurationItems.length; i++) {
-	    
+
         var item = configurationItems[i];
-	    
+
         if (item.resourceType === 'AWS::CloudTrail::Trail') {
-            
+
             if (item.configuration.isMultiRegionTrail) { return 'COMPLIANT'; }
-            
+
         }
-	
+
     }
-    
+
     return 'NON_COMPLIANT';
-    
+
 }
- 
+
 // Helper function used to validate input
- 
+
 function checkDefined(reference, referenceName) {
     if (!reference) {
         console.log("Error: " + referenceName + " is not defined");
         throw referenceName;
     }
     return reference;
 }
- 
+
 // Extract the account ID from the event
- 
+
 function getAccountId(invokingEvent) {
     checkDefined(invokingEvent, "invokingEvent");
     checkDefined(invokingEvent.s3ObjectKey, "invokingEvent.s3ObjectKey");
     var accountIdPattern = /AWSLogs\/(\d+)\/Config/;
     return accountIdPattern.exec(invokingEvent.s3ObjectKey)[1];
 }
- 
+
+// readSnapshot Constructor for retrieving config s3 bucket region
+var readSnapshot = function(s3key, s3bucket, callback) {
+
+    s3.getBucketLocation({"Bucket": s3bucket}, function (err,data){
+        if(err){
+            callback(err,data);
+        } else {
+            var s3region = data.LocationConstraint ? data.LocationConstraint : 'us-east-1';
+            readSnapshot.returnConfig(s3key, s3bucket, s3region, callback);
+        }
+    });
+};
+
 // Reads and parses the ConfigurationSnapshot from the S3 bucket
 // where Config is set up to deliver
- 
-function readSnapshot(s3client, s3key, s3bucket, callback) {
+readSnapshot.returnConfig = function(s3key, s3bucket, s3region, callback){
```

Comment:
This convention ("readSnapshot.returnConfig") is a little unusual. So returnConfig is being scoped to readSnapshot?